### PR TITLE
Still mostly harmless :-)

### PR DESCRIPTION
**Pull request by simogasp**
Main contributions:
* some sparse documentation
* OpenCV is now managed in the main `CMakeLists.txt` instead of having `find_package` for each target that depends on it
* added `OpenMVG_BUILD_DOC` to enable/disable doc building (default: ON)
* some fixes for Clang, in particular for people using the clang version supporting openMP (http://clang-omp.github.io/)
* updated `OptimizeForArchitecture.cmake` with some more recent HW

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -19,10 +19,23 @@ ENDIF()
...
+OPTION(OpenMVG_BUILD_DOC "Build OpenMVG documentation" ON)
 OPTION(OpenMVG_BUILD_EXAMPLES "Build OpenMVG samples applications." ON)
 OPTION(OpenMVG_BUILD_OPENGL_EXAMPLES "Build OpenMVG openGL examples" OFF)
 OPTION(OpenMVG_BUILD_COVERAGE "Enable code coverage generation (gcc only)" OFF)
 OPTION(OpenMVG_USE_OPENMP "Enable OpenMP parallelization" ON)
+# ==============================================================================
+# Opencv is not used by openMVG but some samples show how to use openCV
+#  and openMVG simultaneously
+# ==============================================================================
+OPTION(OpenMVG_USE_OPENCV "Build or not opencv+openMVG samples programs" OFF)
```
Comment (pmoulon) [inline src/CMakeLists.txt]: Later I think I will make the two OpenCV sample external.

File: src/openMVG/sfm/pipelines/localization/SfM_Localizer_Single_3DTrackObservation_Database.cpp [hunk 2]
```diff
@@ -80,16 +80,22 @@ namespace sfm {
...
     matching::IndMatches vec_putative_matches;
     if (!matching_interface_.Match(0.8, query_regions, vec_putative_matches))
     {
+      std::cerr << std::endl
+        << "Matching failed!" << std::endl;
       return false;
     }
 
-    std::cout << "#3D2d putative correspondences: " << vec_putative_matches.size() << std::endl;
+    std::cout << "Number of 3D2d putative correspondences found: "
```
Comment (pmoulon) [inline src/openMVG/sfm/pipelines/localization/SfM_Localizer_Single_3DTrackObservation_Database.cpp]: FYI, I use # to mean cardinal ("number of"), this tip allow to make more compact log.

**Comment by pmoulon** [comment]
It would be better to use git rebase to remove the two commit about -Wno-deprecated-register
Also you can remove the commit related to [travis] (temporaly) re-enabling travis for all branches

**Comment by pmoulon** [comment]
If you cannot do it, I can try to do it.

**Comment by simogasp** [comment]
ok i try to do it later in the afternoon

**Comment by simogasp** [comment]
i did the rebase, my first time, i hope i didn't mess things up... The diff seems ok at least
in the process i also added a missing include in the ui (<clocale>, as usual clang is quite a complainer)

File: src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp [hunk 3]
```diff
@@ -484,7 +484,7 @@ void GlobalSfMReconstructionEngine_RelativeMotions::Compute_Relative_Rotations
   #pragma omp parallel for schedule(dynamic)
 #endif
   // Compute the relative pose from pairwise point matches:
-  for (int i = 0; i < poseWiseMatches.size(); ++i)
+  for (size_t i = 0; i < poseWiseMatches.size(); ++i)
```
Comment (pmoulon) [inline src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp]: size_t will complain on VS ;-)

Comment (simogasp) [inline src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp]: what you mean? It needs std:: in front or it does not exist in VS?

Comment (pmoulon) [inline src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp]: I mean that parallel for in Visual Studio is working only on int type.
When you use size_t it complains.

Comment (simogasp) [inline src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp]: ok sorry it must have slipped in when i did the rebase and solved some conflicts

Comment (simogasp) [inline src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp]: ok it seems to be back to the original commit

**Comment by pmoulon** [comment]
Thanks @simogasp. Merged.

File: src/openMVG/matching/regions_matcher.cpp [hunk 4]
```diff
@@ -158,8 +156,7 @@ Matcher_Regions_Database::Matcher_Regions_Database
       }
     }
   }
-  else
-  if (database_regions.IsBinary() && database_regions.Type_id() == typeid(unsigned char).name())
+  else if (database_regions.IsBinary() and database_regions.Type_id() == typeid(unsigned char).name())
```
Comment (pmoulon) [inline src/openMVG/matching/regions_matcher.cpp]: What the and... (too late it is merged...) Will re-change it.

**Pull request merged**